src/care/config.h.in

```diff
@@ -29,17 +29,21 @@
 #define CARE_DLL_API
 #endif // defined(_WIN32) && !defined(CARESTATICLIB)
 
-#if defined(__CUDACC__)
+#if defined(__CUDACC__) || defined(__HIPCC__)
```

Comment:
I think I would prefer it if you would not combine these because anyone else contributing will be confused by `__CUDACC_OR_HIPCC__`. People know to look for `__CUDACC__` and `__HIPCC__`, but they'll wonder if `__CUDACC_OR_HIPCC__` was set by the compiler or defined by us. It took me a bit to find where this was defined. Also, identifiers beginning with two underscores are generally reserved by the standard/compiler.

Also, you trade brevity for an undesirable pattern in these cases:

```
#ifdef __CUDACC_OR_HIPCC__
#ifdef __CUDACC__
// do something
#else
// do something else
#endif
#endif
```

I'd rather see it as follows:

```
#if defined(__CUDACC__)
// do something
#elif defined(__HIPCC__)
// do something else
#endif 
```

Reply:
I am certainly open to discussion on this, though. I just don't want to introduce something that might confuse people or obscure what is happening. `#if defined(__CUDACC__) || defined(__HIPCC__)' is not that much longer to write out in the cases where it is needed.

Reply:
If you decide to proceed with it as is, at least remove the prefixed and postfixed underscores. Also, just define the variable (don't define it to be 1) so that it is consistent with how we use it (`#if defined` rather than just `#if`).

Reply:
Peter pointed out that using a macro like this reduces the maintenance burden. He suggested and we settled on the name `__GPUCC__`.